byterun/get_set.h

```diff
@@ -0,0 +1,145 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           */
+/*                                                                        */
+/*   Copyright 1996 Institut National de Recherche en Informatique et     */
+/*     en Automatique.                                                    */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+#include "caml/mlvalues.h"
+#include "caml/alloc.h"
+
+static inline value mem_get16(unsigned char* ptr)
+{
+  intnat res;
+  unsigned char b1, b2;
+  b1 = ptr[0];
+  b2 = ptr[1];
+#ifdef ARCH_BIG_ENDIAN
+  res = b1 << 8 | b2;
+#else
+  res = b2 << 8 | b1;
+#endif
+  return Val_int(res);
+}
+
+static inline value mem_get32(unsigned char* ptr)
+{
+  intnat res;
```

Comment:
No I think I didn't consider memcpy. I was young and naive and didn't think that gcc might optimize it.